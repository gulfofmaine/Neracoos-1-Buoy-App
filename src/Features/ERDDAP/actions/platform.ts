import * as Sentry from "@sentry/browser"
import { Action, ActionCreator, Dispatch } from "redux"
import { ThunkAction } from "redux-thunk"

import { StoreState } from "Shared/constants"

import * as actionTypes from "../actionTypes"
import { PlatformFeatureCollection } from "../types"

// Action types
export interface ErddapPlatformLoadSuccess {
  type: actionTypes.ERDDAP_PLATFORM_LOAD_SUCCESS
  geojson: PlatformFeatureCollection
}

export interface ErddapPlatformLoadError {
  type: actionTypes.ERDDAP_PLATFORM_LOAD_ERROR
  message: string
}

export interface ErddapPlatformLoadStarted {
  type: actionTypes.ERDDAP_PLATFORM_LOAD_STARTED
}

export type ErddapPlatformActions = ErddapPlatformLoadSuccess | ErddapPlatformLoadError | ErddapPlatformLoadStarted

// Action creators

/**
 * Action creator for successfully loaded platforms
 * @param geojson GeoJSON feature collection of successfully loaded platforms
 */
export function erddapPlatformLoadSuccess(geojson: PlatformFeatureCollection): ErddapPlatformLoadSuccess {
  return {
    geojson,
    type: actionTypes.ERDDAP_PLATFORM_LOAD_SUCCESS
  }
}

/**
 * Action creator when platforms are unable to be loaded
 * @param message Error message
 */
export function erddapPlatformLoadError(message: string): ErddapPlatformLoadError {
  return {
    message,
    type: actionTypes.ERDDAP_PLATFORM_LOAD_ERROR
  }
}

/**
 * Action creator for when platforms start loading
 */
export function erddapPlatformLoadStarted(): ErddapPlatformLoadStarted {
  return {
    type: actionTypes.ERDDAP_PLATFORM_LOAD_STARTED
  }
}

/**
 * Load platforms from ERDDAP service
 */
export const erddapPlatformLoad: ActionCreator<ThunkAction<Promise<Action>, StoreState, undefined, Action>> = () => {
  return async (dispatch: Dispatch): Promise<Action> => {
    try {
      dispatch(erddapPlatformLoadStarted())

      const url = (process.env.REACT_APP_ERDDAP_SERVICE as string) + "/api/platforms/"

      Sentry.addBreadcrumb({
        category: "ERDDAP Service",
        data: {
          url
        },
        message: "Loading platform GeoJSON"
      })

      const result = await fetch(url)
      const json = (await result.json()) as PlatformFeatureCollection

      return dispatch(erddapPlatformLoadSuccess(json))
    } catch (error) {
      // tslint:disable-next-line:no-console
      // console.log(error)
      Sentry.captureException(error)

      return dispatch(erddapPlatformLoadError("Unable to load platform data"))
    }
  }
}
