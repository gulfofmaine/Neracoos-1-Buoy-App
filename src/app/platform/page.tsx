import { Col, Row } from "components/Bootstrap/layout"
import React from "react"

export default function PlatformLayout() {
  return (
    <Row>
      <Col sm={{ size: true, order: 2 }}>Platform list</Col>
      <Col sm={{ size: true, order: 1 }}>Map</Col>
    </Row>
  )
}
