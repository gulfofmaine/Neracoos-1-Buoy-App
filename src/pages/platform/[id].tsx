import { GetStaticPaths, GetStaticProps } from "next"
import React from "react"
import { QueryCache } from "react-query"
import { dehydrate } from "react-query/hydration"

import { PlatformLayout } from "components/Layout"
import { ErddapCurrentPlatformConditions, BUOY_BARN_PLATFORMS_KEY, getPlatforms } from "Features/ERDDAP"

const CurrentConditions: React.FunctionComponent = () => {
  return <PlatformLayout>{({ platform }) => <ErddapCurrentPlatformConditions platform={platform} />}</PlatformLayout>
}

export const getStaticProps: GetStaticProps = async () => {
  const queryCache = new QueryCache()

  await queryCache.prefetchQuery(BUOY_BARN_PLATFORMS_KEY, getPlatforms)

  return {
    props: {
      dehydratedState: dehydrate(queryCache),
    },
    revalidate: 10 * 60, // Every ten minutes
  }
}

export const getStaticPaths: GetStaticPaths = async () => {
  const platforms = await getPlatforms()

  const paths = platforms.features.map((platform) => ({ params: { id: platform.id } }))

  return {
    paths,
    fallback: true,
  }
}

export default CurrentConditions
