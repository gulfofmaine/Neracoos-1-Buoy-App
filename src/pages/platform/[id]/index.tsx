import { GetStaticPaths, GetStaticProps } from "next"
import React from "react"
import { QueryCache } from "react-query"
import { dehydrate } from "react-query/hydration"
import { Col, Row } from "reactstrap"

import { PlatformLayout } from "components/Layout"
import {
  ErddapCurrentPlatformConditions,
  BUOY_BARN_PLATFORMS_KEY,
  getPlatforms,
  BUOY_BARN_FORECAST_KEY,
  getForecasts,
} from "Features/ERDDAP"

const CurrentConditions: React.FunctionComponent = () => {
  return (
    <PlatformLayout>
      {({ platform }) => (
        <Row id="currentConditions">
          <Col>
            <ErddapCurrentPlatformConditions platform={platform} />
          </Col>
        </Row>
      )}
    </PlatformLayout>
  )
}

export const getStaticProps: GetStaticProps = async () => {
  const queryCache = new QueryCache()

  await queryCache.prefetchQuery(BUOY_BARN_PLATFORMS_KEY, getPlatforms)
  await queryCache.prefetchQuery(BUOY_BARN_FORECAST_KEY, getForecasts)

  return {
    props: {
      dehydratedState: dehydrate(queryCache),
    },
    revalidate: 10 * 60, // Every ten minutes
  }
}

export const getStaticPaths: GetStaticPaths = async () => {
  return {
    paths: [],
    fallback: true,
  }
}

export default CurrentConditions
