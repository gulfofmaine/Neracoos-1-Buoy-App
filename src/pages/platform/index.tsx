import { GetStaticProps } from "next"
import { useRouter } from "next/router"
import React from "react"
import { QueryCache } from "react-query"
import { dehydrate } from "react-query/hydration"

import { MapLayout } from "components/Layout"
import { ErddapPlatformList, BUOY_BARN_PLATFORMS_KEY, getPlatforms } from "Features/ERDDAP"
import { Region } from "Shared/regions"
import { regionList } from "Shared/constants"

const PlatformPage: React.FunctionComponent = () => {
  const { query } = useRouter()

  let region: Region | undefined

  if (query.region) {
    region = regionList.find((r) => r.slug === query.region)
  }

  return (
    <MapLayout
      pageName={region?.name}
      boundingBox={region?.bbox}
      rightCol={
        <React.Fragment>
          <h2>Platforms in {region ? region.name : ""}</h2>
          <ErddapPlatformList boundingBox={region?.bbox} />
        </React.Fragment>
      }
    >
      {null}
    </MapLayout>
  )
}

export const getStaticProps: GetStaticProps = async () => {
  const queryCache = new QueryCache()

  await queryCache.prefetchQuery(BUOY_BARN_PLATFORMS_KEY, getPlatforms)

  return {
    props: {
      dehydratedState: dehydrate(queryCache),
    },
    revalidate: 10 * 60, // Every ten minutes
  }
}

export default PlatformPage
