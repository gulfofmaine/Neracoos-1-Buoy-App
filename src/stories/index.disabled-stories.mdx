import { Meta } from "@storybook/addon-docs/blocks"

<Meta title="Mariner's Dashboard" />

# Mariner's Dashboard

The [NERACOOS Mariner's Dashboard](https://mariners.neracoos.org/)
is the next generation of our buoy map, redesigned to work natively
with [ERDDAP](http://www.neracoos.org/erddap/index.html) as a
backend data store.

A lightweight service ([Buoy Barn](https://github.com/gulfofmaine/buoy_barn))
provides data about platforms and data to display, latest values,
and enough information that the Mariner's Dashboard is able to
query ERDDAP itself.

## Components and dashboard structure

Mariner's Dashboard is organized around three key functionalities.

- Retrieving and displaying data from ERDDAP.
- User selection of display units.
- Retrieving and displaying the home and about pages.

More specialized components are organized around those categories,
while more general ones are under **Components**.
