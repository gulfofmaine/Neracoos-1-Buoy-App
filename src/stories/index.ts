import { storiesOf } from '@storybook/angular'
// import { action } from '@storybook/addon-actions'
import { MiniMapComponent } from '../components/mini-map/mini-map';

storiesOf('Mini Map', module)
    .add('Default', () => ({
        component: MiniMapComponent,
        props: {},
        moduleMetadata: {
            providers: [
                
            ]
        }
    }))
