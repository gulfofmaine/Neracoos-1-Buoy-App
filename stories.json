{"v":3,"stories":{"erddap-forecast-configurable--docs":{"id":"erddap-forecast-configurable--docs","title":"ERDDAP/Forecast/Configurable","name":"Docs","importPath":"./src/Features/ERDDAP/Platform/Forecasts/Page/index-configurable.stories.tsx","tags":["docs","autodocs"],"storiesImports":[],"kind":"ERDDAP/Forecast/Configurable","story":"Docs","parameters":{"__id":"erddap-forecast-configurable--docs","docsOnly":true,"fileName":"./src/Features/ERDDAP/Platform/Forecasts/Page/index-configurable.stories.tsx"}},"erddap-forecast-configurable--english":{"id":"erddap-forecast-configurable--english","title":"ERDDAP/Forecast/Configurable","name":"English","importPath":"./src/Features/ERDDAP/Platform/Forecasts/Page/index-configurable.stories.tsx","tags":["story"],"kind":"ERDDAP/Forecast/Configurable","story":"English","parameters":{"__id":"erddap-forecast-configurable--english","docsOnly":false,"fileName":"./src/Features/ERDDAP/Platform/Forecasts/Page/index-configurable.stories.tsx"}},"erddap-forecast-configurable--metric":{"id":"erddap-forecast-configurable--metric","title":"ERDDAP/Forecast/Configurable","name":"Metric","importPath":"./src/Features/ERDDAP/Platform/Forecasts/Page/index-configurable.stories.tsx","tags":["story"],"kind":"ERDDAP/Forecast/Configurable","story":"Metric","parameters":{"__id":"erddap-forecast-configurable--metric","docsOnly":false,"fileName":"./src/Features/ERDDAP/Platform/Forecasts/Page/index-configurable.stories.tsx"}},"erddap-info-panel--docs":{"id":"erddap-info-panel--docs","title":"ERDDAP/Info/Panel","name":"Docs","importPath":"./src/Features/ERDDAP/Platform/Info/index.stories.tsx","tags":["docs","autodocs"],"storiesImports":[],"kind":"ERDDAP/Info/Panel","story":"Docs","parameters":{"__id":"erddap-info-panel--docs","docsOnly":true,"fileName":"./src/Features/ERDDAP/Platform/Info/index.stories.tsx"}},"erddap-info-panel--platform-info":{"id":"erddap-info-panel--platform-info","title":"ERDDAP/Info/Panel","name":"Platform Info","importPath":"./src/Features/ERDDAP/Platform/Info/index.stories.tsx","tags":["story"],"kind":"ERDDAP/Info/Panel","story":"Platform Info","parameters":{"__id":"erddap-info-panel--platform-info","docsOnly":false,"fileName":"./src/Features/ERDDAP/Platform/Info/index.stories.tsx"}},"erddap-observedcondition-configurable--docs":{"id":"erddap-observedcondition-configurable--docs","title":"ERDDAP/ObservedCondition/Configurable","name":"Docs","importPath":"./src/Features/ERDDAP/Platform/Observations/Condition/index-configurable.stories.tsx","tags":["docs","autodocs"],"storiesImports":[],"kind":"ERDDAP/ObservedCondition/Configurable","story":"Docs","parameters":{"__id":"erddap-observedcondition-configurable--docs","docsOnly":true,"fileName":"./src/Features/ERDDAP/Platform/Observations/Condition/index-configurable.stories.tsx"}},"erddap-observedcondition-configurable--english":{"id":"erddap-observedcondition-configurable--english","title":"ERDDAP/ObservedCondition/Configurable","name":"English","importPath":"./src/Features/ERDDAP/Platform/Observations/Condition/index-configurable.stories.tsx","tags":["story"],"kind":"ERDDAP/ObservedCondition/Configurable","story":"English","parameters":{"__id":"erddap-observedcondition-configurable--english","docsOnly":false,"fileName":"./src/Features/ERDDAP/Platform/Observations/Condition/index-configurable.stories.tsx"}},"erddap-currentconditions-data-card--docs":{"id":"erddap-currentconditions-data-card--docs","title":"ERDDAP/CurrentConditions/Data Card","name":"Docs","importPath":"./src/Features/ERDDAP/Platform/Observations/CurrentConditions/data_card.stories.tsx","tags":["docs","autodocs"],"storiesImports":[],"kind":"ERDDAP/CurrentConditions/Data Card","story":"Docs","parameters":{"__id":"erddap-currentconditions-data-card--docs","docsOnly":true,"fileName":"./src/Features/ERDDAP/Platform/Observations/CurrentConditions/data_card.stories.tsx"}},"erddap-currentconditions-data-card--configurable":{"id":"erddap-currentconditions-data-card--configurable","title":"ERDDAP/CurrentConditions/Data Card","name":"Configurable","importPath":"./src/Features/ERDDAP/Platform/Observations/CurrentConditions/data_card.stories.tsx","tags":["story"],"kind":"ERDDAP/CurrentConditions/Data Card","story":"Configurable","parameters":{"__id":"erddap-currentconditions-data-card--configurable","docsOnly":false,"fileName":"./src/Features/ERDDAP/Platform/Observations/CurrentConditions/data_card.stories.tsx"}},"erddap-observationtable-all--docs":{"id":"erddap-observationtable-all--docs","title":"ERDDAP/ObservationTable/All","name":"Docs","importPath":"./src/Features/ERDDAP/Platform/Observations/Table/all.stories.tsx","tags":["docs","autodocs"],"storiesImports":[],"kind":"ERDDAP/ObservationTable/All","story":"Docs","parameters":{"__id":"erddap-observationtable-all--docs","docsOnly":true,"fileName":"./src/Features/ERDDAP/Platform/Observations/Table/all.stories.tsx"}},"erddap-observationtable-all--english":{"id":"erddap-observationtable-all--english","title":"ERDDAP/ObservationTable/All","name":"English","importPath":"./src/Features/ERDDAP/Platform/Observations/Table/all.stories.tsx","tags":["story"],"kind":"ERDDAP/ObservationTable/All","story":"English","parameters":{"__id":"erddap-observationtable-all--english","docsOnly":false,"fileName":"./src/Features/ERDDAP/Platform/Observations/Table/all.stories.tsx"}},"erddap-observationtable-all--metric":{"id":"erddap-observationtable-all--metric","title":"ERDDAP/ObservationTable/All","name":"Metric","importPath":"./src/Features/ERDDAP/Platform/Observations/Table/all.stories.tsx","tags":["story"],"kind":"ERDDAP/ObservationTable/All","story":"Metric","parameters":{"__id":"erddap-observationtable-all--metric","docsOnly":false,"fileName":"./src/Features/ERDDAP/Platform/Observations/Table/all.stories.tsx"}},"erddap-observationtable-item--docs":{"id":"erddap-observationtable-item--docs","title":"ERDDAP/ObservationTable/Item","name":"Docs","importPath":"./src/Features/ERDDAP/Platform/Observations/Table/item.stories.tsx","tags":["docs","autodocs"],"storiesImports":[],"kind":"ERDDAP/ObservationTable/Item","story":"Docs","parameters":{"__id":"erddap-observationtable-item--docs","docsOnly":true,"fileName":"./src/Features/ERDDAP/Platform/Observations/Table/item.stories.tsx"}},"erddap-observationtable-item--english":{"id":"erddap-observationtable-item--english","title":"ERDDAP/ObservationTable/Item","name":"English","importPath":"./src/Features/ERDDAP/Platform/Observations/Table/item.stories.tsx","tags":["story"],"kind":"ERDDAP/ObservationTable/Item","story":"English","parameters":{"__id":"erddap-observationtable-item--english","docsOnly":false,"fileName":"./src/Features/ERDDAP/Platform/Observations/Table/item.stories.tsx"}},"erddap-observationtable-item--metric":{"id":"erddap-observationtable-item--metric","title":"ERDDAP/ObservationTable/Item","name":"Metric","importPath":"./src/Features/ERDDAP/Platform/Observations/Table/item.stories.tsx","tags":["story"],"kind":"ERDDAP/ObservationTable/Item","story":"Metric","parameters":{"__id":"erddap-observationtable-item--metric","docsOnly":false,"fileName":"./src/Features/ERDDAP/Platform/Observations/Table/item.stories.tsx"}},"erddap-observationtable-current--docs":{"id":"erddap-observationtable-current--docs","title":"ERDDAP/ObservationTable/Current","name":"Docs","importPath":"./src/Features/ERDDAP/Platform/Observations/Table/table.stories.tsx","tags":["docs","autodocs"],"storiesImports":[],"kind":"ERDDAP/ObservationTable/Current","story":"Docs","parameters":{"__id":"erddap-observationtable-current--docs","docsOnly":true,"fileName":"./src/Features/ERDDAP/Platform/Observations/Table/table.stories.tsx"}},"erddap-observationtable-current--english":{"id":"erddap-observationtable-current--english","title":"ERDDAP/ObservationTable/Current","name":"English","importPath":"./src/Features/ERDDAP/Platform/Observations/Table/table.stories.tsx","tags":["story"],"kind":"ERDDAP/ObservationTable/Current","story":"English","parameters":{"__id":"erddap-observationtable-current--english","docsOnly":false,"fileName":"./src/Features/ERDDAP/Platform/Observations/Table/table.stories.tsx"}},"erddap-observationtable-current--metric":{"id":"erddap-observationtable-current--metric","title":"ERDDAP/ObservationTable/Current","name":"Metric","importPath":"./src/Features/ERDDAP/Platform/Observations/Table/table.stories.tsx","tags":["story"],"kind":"ERDDAP/ObservationTable/Current","story":"Metric","parameters":{"__id":"erddap-observationtable-current--metric","docsOnly":false,"fileName":"./src/Features/ERDDAP/Platform/Observations/Table/table.stories.tsx"}},"erddap-superlatives--docs":{"id":"erddap-superlatives--docs","title":"ERDDAP/Superlatives","name":"Docs","importPath":"./src/Features/ERDDAP/Superlatives/index.stories.tsx","tags":["docs","autodocs"],"storiesImports":[],"kind":"ERDDAP/Superlatives","story":"Docs","parameters":{"__id":"erddap-superlatives--docs","docsOnly":true,"fileName":"./src/Features/ERDDAP/Superlatives/index.stories.tsx"}},"erddap-superlatives--configureable":{"id":"erddap-superlatives--configureable","title":"ERDDAP/Superlatives","name":"Configureable","importPath":"./src/Features/ERDDAP/Superlatives/index.stories.tsx","tags":["story"],"kind":"ERDDAP/Superlatives","story":"Configureable","parameters":{"__id":"erddap-superlatives--configureable","docsOnly":false,"fileName":"./src/Features/ERDDAP/Superlatives/index.stories.tsx"}},"erddap-superlatives--should-not-show-outdated-data":{"id":"erddap-superlatives--should-not-show-outdated-data","title":"ERDDAP/Superlatives","name":"Should Not Show Outdated Data","importPath":"./src/Features/ERDDAP/Superlatives/index.stories.tsx","tags":["story"],"kind":"ERDDAP/Superlatives","story":"Should Not Show Outdated Data","parameters":{"__id":"erddap-superlatives--should-not-show-outdated-data","docsOnly":false,"fileName":"./src/Features/ERDDAP/Superlatives/index.stories.tsx"}},"units-unitselector-configurable--docs":{"id":"units-unitselector-configurable--docs","title":"Units/UnitSelector/configurable","name":"Docs","importPath":"./src/Features/Units/UnitSelector/index-configurable.stories.tsx","tags":["docs","autodocs"],"storiesImports":[],"kind":"Units/UnitSelector/configurable","story":"Docs","parameters":{"__id":"units-unitselector-configurable--docs","docsOnly":true,"fileName":"./src/Features/Units/UnitSelector/index-configurable.stories.tsx"}},"units-unitselector-configurable--english":{"id":"units-unitselector-configurable--english","title":"Units/UnitSelector/configurable","name":"English","importPath":"./src/Features/Units/UnitSelector/index-configurable.stories.tsx","tags":["story"],"kind":"Units/UnitSelector/configurable","story":"English","parameters":{"__id":"units-unitselector-configurable--english","docsOnly":false,"fileName":"./src/Features/Units/UnitSelector/index-configurable.stories.tsx"}},"units-unitselector-configurable--metric":{"id":"units-unitselector-configurable--metric","title":"Units/UnitSelector/configurable","name":"Metric","importPath":"./src/Features/Units/UnitSelector/index-configurable.stories.tsx","tags":["story"],"kind":"Units/UnitSelector/configurable","story":"Metric","parameters":{"__id":"units-unitselector-configurable--metric","docsOnly":false,"fileName":"./src/Features/Units/UnitSelector/index-configurable.stories.tsx"}},"components-500-page--docs":{"id":"components-500-page--docs","title":"Components/500 Page","name":"Docs","importPath":"./src/Pages/500/index.stories.tsx","tags":["docs","autodocs"],"storiesImports":[],"kind":"Components/500 Page","story":"Docs","parameters":{"__id":"components-500-page--docs","docsOnly":true,"fileName":"./src/Pages/500/index.stories.tsx"}},"components-500-page--five-hundred-page":{"id":"components-500-page--five-hundred-page","title":"Components/500 Page","name":"Five Hundred Page","importPath":"./src/Pages/500/index.stories.tsx","tags":["story"],"kind":"Components/500 Page","story":"Five Hundred Page","parameters":{"__id":"components-500-page--five-hundred-page","docsOnly":false,"fileName":"./src/Pages/500/index.stories.tsx"}},"modeling-chart--docs":{"id":"modeling-chart--docs","title":"Modeling/Chart","name":"Docs","importPath":"./src/Pages/Modeling/chart.stories.tsx","tags":["docs","autodocs"],"storiesImports":[],"kind":"Modeling/Chart","story":"Docs","parameters":{"__id":"modeling-chart--docs","docsOnly":true,"fileName":"./src/Pages/Modeling/chart.stories.tsx"}},"modeling-chart--chart":{"id":"modeling-chart--chart","title":"Modeling/Chart","name":"Chart","importPath":"./src/Pages/Modeling/chart.stories.tsx","tags":["story"],"kind":"Modeling/Chart","story":"Chart","parameters":{"__id":"modeling-chart--chart","docsOnly":false,"fileName":"./src/Pages/Modeling/chart.stories.tsx"}},"modeling-table--docs":{"id":"modeling-table--docs","title":"Modeling/Table","name":"Docs","importPath":"./src/Pages/Modeling/table.stories.tsx","tags":["docs","autodocs"],"storiesImports":[],"kind":"Modeling/Table","story":"Docs","parameters":{"__id":"modeling-table--docs","docsOnly":true,"fileName":"./src/Pages/Modeling/table.stories.tsx"}},"modeling-table--table":{"id":"modeling-table--table","title":"Modeling/Table","name":"Table","importPath":"./src/Pages/Modeling/table.stories.tsx","tags":["story"],"kind":"Modeling/Table","story":"Table","parameters":{"__id":"modeling-table--table","docsOnly":false,"fileName":"./src/Pages/Modeling/table.stories.tsx"}},"components-charts-windtimeserieschart--docs":{"id":"components-charts-windtimeserieschart--docs","title":"Components/Charts/WindTimeSeriesChart","name":"Docs","importPath":"./src/components/Charts/WindTimeSeries.stories.tsx","tags":["docs","autodocs"],"storiesImports":[],"kind":"Components/Charts/WindTimeSeriesChart","story":"Docs","parameters":{"__id":"components-charts-windtimeserieschart--docs","docsOnly":true,"fileName":"./src/components/Charts/WindTimeSeries.stories.tsx"}},"components-charts-windtimeserieschart--doesnt-explode-with-negative-dates":{"id":"components-charts-windtimeserieschart--doesnt-explode-with-negative-dates","title":"Components/Charts/WindTimeSeriesChart","name":"Doesnt Explode With Negative Dates","importPath":"./src/components/Charts/WindTimeSeries.stories.tsx","tags":["story"],"kind":"Components/Charts/WindTimeSeriesChart","story":"Doesnt Explode With Negative Dates","parameters":{"__id":"components-charts-windtimeserieschart--doesnt-explode-with-negative-dates","docsOnly":false,"fileName":"./src/components/Charts/WindTimeSeries.stories.tsx"}},"components-charts-windtimeserieschart--scale-should-reach-40":{"id":"components-charts-windtimeserieschart--scale-should-reach-40","title":"Components/Charts/WindTimeSeriesChart","name":"Scale Should Reach 40","importPath":"./src/components/Charts/WindTimeSeries.stories.tsx","tags":["story"],"kind":"Components/Charts/WindTimeSeriesChart","story":"Scale Should Reach 40","parameters":{"__id":"components-charts-windtimeserieschart--scale-should-reach-40","docsOnly":false,"fileName":"./src/components/Charts/WindTimeSeries.stories.tsx"}},"components-footer--docs":{"id":"components-footer--docs","title":"Components/Footer","name":"Docs","importPath":"./src/components/Footer/index.stories.tsx","tags":["docs","autodocs"],"storiesImports":[],"kind":"Components/Footer","story":"Docs","parameters":{"__id":"components-footer--docs","docsOnly":true,"fileName":"./src/components/Footer/index.stories.tsx"}},"components-footer--footer":{"id":"components-footer--footer","title":"Components/Footer","name":"Footer","importPath":"./src/components/Footer/index.stories.tsx","tags":["story"],"kind":"Components/Footer","story":"Footer","parameters":{"__id":"components-footer--footer","docsOnly":false,"fileName":"./src/components/Footer/index.stories.tsx"}},"components-navbar--docs":{"id":"components-navbar--docs","title":"Components/NavBar","name":"Docs","importPath":"./src/components/NavBar/index.stories.tsx","tags":["docs","autodocs"],"storiesImports":[],"kind":"Components/NavBar","story":"Docs","parameters":{"__id":"components-navbar--docs","docsOnly":true,"fileName":"./src/components/NavBar/index.stories.tsx"}},"components-navbar--nav-bar":{"id":"components-navbar--nav-bar","title":"Components/NavBar","name":"Nav Bar","importPath":"./src/components/NavBar/index.stories.tsx","tags":["story"],"kind":"Components/NavBar","story":"Nav Bar","parameters":{"__id":"components-navbar--nav-bar","docsOnly":false,"fileName":"./src/components/NavBar/index.stories.tsx"}},"erddap-forecast--docs":{"id":"erddap-forecast--docs","title":"ERDDAP/Forecast","name":"Docs","importPath":"./src/Features/ERDDAP/Platform/Forecasts/Page/index.stories.mdx","tags":["stories-mdx","docs","autodocs"],"storiesImports":[],"kind":"ERDDAP/Forecast","story":"Docs","parameters":{"__id":"erddap-forecast--docs","docsOnly":true,"fileName":"./src/Features/ERDDAP/Platform/Forecasts/Page/index.stories.mdx"}},"erddap-forecast--forecast-chart":{"id":"erddap-forecast--forecast-chart","title":"ERDDAP/Forecast","name":"Forecast Chart","importPath":"./src/Features/ERDDAP/Platform/Forecasts/Page/index.stories.mdx","tags":["stories-mdx","story"],"kind":"ERDDAP/Forecast","story":"Forecast Chart","parameters":{"__id":"erddap-forecast--forecast-chart","docsOnly":false,"fileName":"./src/Features/ERDDAP/Platform/Forecasts/Page/index.stories.mdx"}},"erddap-observationtable--docs":{"id":"erddap-observationtable--docs","title":"ERDDAP/ObservationTable","name":"Docs","importPath":"./src/Features/ERDDAP/Platform/Observations/Table/index.stories.mdx","tags":["stories-mdx","docs","autodocs"],"storiesImports":[],"kind":"ERDDAP/ObservationTable","story":"Docs","parameters":{"__id":"erddap-observationtable--docs","docsOnly":true,"fileName":"./src/Features/ERDDAP/Platform/Observations/Table/index.stories.mdx"}},"erddap-observationtable--current-observations":{"id":"erddap-observationtable--current-observations","title":"ERDDAP/ObservationTable","name":"Current Observations","importPath":"./src/Features/ERDDAP/Platform/Observations/Table/index.stories.mdx","tags":["stories-mdx","story"],"kind":"ERDDAP/ObservationTable","story":"Current Observations","parameters":{"__id":"erddap-observationtable--current-observations","docsOnly":false,"fileName":"./src/Features/ERDDAP/Platform/Observations/Table/index.stories.mdx"}},"erddap-observationtable--all-observations":{"id":"erddap-observationtable--all-observations","title":"ERDDAP/ObservationTable","name":"All Observations","importPath":"./src/Features/ERDDAP/Platform/Observations/Table/index.stories.mdx","tags":["stories-mdx","story"],"kind":"ERDDAP/ObservationTable","story":"All Observations","parameters":{"__id":"erddap-observationtable--all-observations","docsOnly":false,"fileName":"./src/Features/ERDDAP/Platform/Observations/Table/index.stories.mdx"}},"units-unitselector--docs":{"id":"units-unitselector--docs","title":"Units/UnitSelector","name":"Docs","importPath":"./src/Features/Units/UnitSelector/index.stories.mdx","tags":["stories-mdx","docs","autodocs"],"storiesImports":[],"kind":"Units/UnitSelector","story":"Docs","parameters":{"__id":"units-unitselector--docs","docsOnly":true,"fileName":"./src/Features/Units/UnitSelector/index.stories.mdx"}},"units-unitselector--unit-selector":{"id":"units-unitselector--unit-selector","title":"Units/UnitSelector","name":"Unit Selector","importPath":"./src/Features/Units/UnitSelector/index.stories.mdx","tags":["stories-mdx","story"],"kind":"Units/UnitSelector","story":"Unit Selector","parameters":{"__id":"units-unitselector--unit-selector","docsOnly":false,"fileName":"./src/Features/Units/UnitSelector/index.stories.mdx"}},"mariner-s-dashboard--docs":{"id":"mariner-s-dashboard--docs","title":"Mariner's Dashboard","name":"Docs","importPath":"./src/stories/index.stories.mdx","tags":["stories-mdx","docs","autodocs"],"storiesImports":[],"kind":"Mariner's Dashboard","story":"Docs","parameters":{"__id":"mariner-s-dashboard--docs","docsOnly":true,"fileName":"./src/stories/index.stories.mdx"}}}}
