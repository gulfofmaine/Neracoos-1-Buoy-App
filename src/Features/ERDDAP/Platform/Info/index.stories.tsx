import * as React from "react"

import { ErddapPlatformInfoPanel } from "./index"

import { platform } from "stories/platform"

export default {
  component: ErddapPlatformInfoPanel,
  title: "ERDDAP/Info/Panel",
}

export const platformInfo = () => <ErddapPlatformInfoPanel platform={platform} />
