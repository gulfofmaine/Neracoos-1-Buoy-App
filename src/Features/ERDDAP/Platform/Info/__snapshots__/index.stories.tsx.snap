// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ERDDAP/Info/Panel platformInfo smoke-test 1`] = `
<div role="complementary"
     class="card"
>
  <div class="card-body">
    <div role="header"
         class="card-title"
    >
      Station M01
    </div>
    <p class="card-text">
      Jordan Basin
      <br>
      <b>
        NDBC ID:
      </b>
      44037
      <br>
      <b>
        Lat:
      </b>
      43.5
      <b>
        Lon:
      </b>
      -67.87
      <br>
      <a href="http://gyre.umeoce.maine.edu/gomoos.php">
        University of Maine
      </a>
      <br>
    </p>
  </div>
</div>
`;
